rasa/cli/data.py

```diff
@@ -91,22 +133,50 @@ def split_nlu_data(args) -> None:
     test.persist(args.out, filename=f"test_data.{fformat}")
 
 
-def validate_files(args) -> NoReturn:
-    """Validate all files needed for training a model.
-
-    Fails with a non-zero exit code if there are any errors in the data."""
-    from rasa.core.validator import Validator
-    from rasa.importers.rasa import RasaFileImporter
-
+def validate_files(args: argparse.Namespace) -> None:
     loop = asyncio.get_event_loop()
     file_importer = RasaFileImporter(
         domain_path=args.domain, training_data_paths=args.data
     )
 
     validator = loop.run_until_complete(Validator.from_importer(file_importer))
-    domain_is_valid = validator.verify_domain_validity()
-    if not domain_is_valid:
+
+    if "stories_only" in args:
+        all_good = _validate_story_structure(validator, args)
+    elif "max_history" not in args or args.max_history is None:
+        logger.info(
+            "Will not test for inconsistencies in stories since "
+            "you did not provide a value for `--max-history`."
+        )
+        all_good = _validate_domain(validator) and _validate_nlu(validator, args)
+    else:
+        all_good = (
+            _validate_domain(validator)
+            and _validate_nlu(validator, args)
+            and _validate_story_structure(validator, args)
```

Comment:
Can we simplify the if else here?

```python

# validate_files:
if args.stories_only:
   validate_story_structure
else:
  validate_all_three(...) # too lazy to list all three functions here

# And then in `_validate_story_structure`
def _validate_story_structure(...):
   if not args.max_history or args.max_history < 1:
       logger.warn("skipping story structure validation. invalid parameter")
       return True

   verify_story_structure(...)
```

Reply:
But while it is ok for the user to omit the `--max-history` setting in `rasa data validate`, it is _not_ ok for him to omit the `--max-history` in `rasa data validate stories`. So I could write
```
all_good = (
                _validate_domain(validator)
                and _validate_nlu(validator, args)
                and (not args.max_history or _validate_story_structure(validator, args))
        )
```
but then there would be no info message that stories aren't been tested. Alternatively, one could add another parameter to `_validate_story_structure`, like `force`, but I'd find that even more confusing. What do you think?

Reply:
@wochinge What do you think?